### ISSUE TITLE
Feature request: Gamecube controller support

### ISSUE THREAD

**Issue by Leon422** [sm64pc/sm64pc#157]
SDL supports GameCube controller adapters. Some software such as Dolphin and URDE support it in both Windows and Linux. Apparently the old leaked version of the port supported GameCube controller adapters (but only on Linux). It would be cool to see the support return, except this time for both OSs :)

**Comment by fgsfdsfgs** [comment]
The "old" version had a controller module that manually polled the adapter using libusb. SDL2 is *supposed to* support the adapter, but most builds of it that I've seen around do not for some reason. Might need to rebuild it with `--enable-hidapi`.

**Comment by Leon422** [comment]
> The "old" version had a controller module that manually polled the adapter using libusb. SDL2 is _supposed to_ support the adapter, but most builds of it that I've seen around do not for some reason. Might need to rebuild it with `--enable-hidapi`.

Pretty sure the libUSB thing is what Dolphin does. I'd venture to say that everyone who uses a GameCube adapter on PC has already set it up to work in Dolphin, so if sm64pc handled it the same way I think no one would complain.

**Comment by uncletrunks** [comment]
In my experience Gamecube controller support has been in all the builds I've done on Arch, with my Mayflash Wii U adapter. For the time being if it's not working on Windows through SDL you might want to try mapping it using Steam.

**Comment by Leon422** [comment]
> 
> 
> In my experience Gamecube controller support has been in all the builds I've done on Arch, with my Mayflash Wii U adapter. For the time being if it's not working on Windows through SDL you might want to try mapping it using Steam.

Yeah I usually use Steam for stuff that doesn't support the controller natively. It's just that the Steam solution is more messy because you have to emulate a virtual controller via vJoy and some times it doesn't feel like working and ugh

**Comment by Sukid** [comment]
So I've been trying to figure this out myself. I can't really figure out if the issue here is SDL2 or something else. What's odd is that it DOES work in Retroarch with sdl2 set as the gamepad driver, but nowhere else (other than Dolphin obviously).
